jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -3219,9 +3228,12 @@ private static final boolean hasMetaChars(String s) {
 	 * @since 1.4
 	 */
 	public String[] split(String regex, int max) {
-		// it is faster to handle simple splits inline (i.e. no fancy regex matching)
+		// it is faster to handle simple splits inline (i.e. no fancy regex matching),
+		// including single escaped literal character (e.g. \. \{),
 		// so we test for a suitable string and handle this here if we can
-		if (regex != null && regex.lengthInternal() > 0 && !hasMetaChars(regex)) {
+		boolean singleEscapeLiteral = false;
+		if (regex != null && regex.lengthInternal() > 0 
+				&& (!hasMetaChars(regex) || (singleEscapeLiteral = isSingleEscapeLiteral(regex)))) {
```

Comment:
`singleEscapeLiteral = isSingleEscapeLiteral(regex`

Please refactor this code so that the assignment isn't done as a side effect of the if statement.  It makes it very difficult to reason about the code when if conditions become overly complex. 

Better to use multiple if statements then to end up with code like this.

Reply:
Please also bracket subexpressions so the operation is clear